### ISSUE TITLE
no-unused-vars issue is erroneously generated for loop variables

### ISSUE THREAD

**Issue by mykola-mokhnach** [eslint/eslint#13947]
**Tell us about your environment**

* **ESLint Version:** 7.16.0
* **Node Version:** 12.20.0
* **npm Version:** 6.4.19

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

babel-eslint@10.0.2

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": "appium"
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
  emitLines (stream, lines) {
    for (let line of lines) {
      this.emit('stream-line', `[${stream.toUpperCase()}] ${line}`);
    }
  }
```

```bash
gulp eslint
```

**What did you expect to happen?**
No linting errors after executing `npm i && gulp eslint` on https://github.com/appium/node-teen_process NPM module

**What actually happened? Please include the actual, raw output from ESLint.**
Before the recent update the code of https://github.com/appium/node-teen_process/blob/master/lib/subprocess.js file was linted without issues. Although, after updating ESLint to 7.16.0 it started to fail with the following errors
```
npm ERR! /Users/elf/code/node-teen_process/lib/subprocess.js
npm ERR!    48:14  error  'line' is defined but never used        no-unused-vars
npm ERR!   123:21  error  'streamName' is defined but never used  no-unused-vars
npm ERR!   123:33  error  'streamData' is defined but never used  no-unused-vars
npm ERR!   211:14  error  'stream' is defined but never used      no-unused-vars
npm ERR! 
npm ERR! ✖ 4 problems (4 errors, 0 warnings)
```

After checking the corresponding code it turns out these errors are invalid, because the variables are actually used inside their loops. 

**Are you willing to submit a pull request to fix this bug?**

If I only knew how

**Comment by anikethsaha** [comment]
THanks for the issue. 


It is working fine with the mentioned version with default parser. [demo](https://eslint.org/demo#eyJ0ZXh0IjoiY2xhc3MgQSB7XG4gIGVtaXRMaW5lcyAoc3RyZWFtLCBsaW5lcykge1xuICAgIGZvciAobGV0IGxpbmUgb2YgbGluZXMpIHtcbiAgICAgIHRoaXMuZW1pdCgnc3RyZWFtLWxpbmUnLCBgWyR7c3RyZWFtLnRvVXBwZXJDYXNlKCl9XSAke2xpbmV9YCk7XG4gICAgfVxuICB9XG5cblxufVxuXG5uZXcgQSgpIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJuby11bnVzZWQtdmFycyI6Mn0sImVudiI6e319fQ==)

Make sure the dependencies are updated properly. Looks like the appium config is not updated for a while.  also use the latest `@babel/eslint-parser` if using latest eslint.

**Comment by mykola-mokhnach** [comment]
Thanks @anikethsaha 
Let us try to update eslint-parser first

**Comment by mykola-mokhnach** [comment]
Bumping @bable/eslint-parser has solved the issue. Thanks.
